src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
> Hopefully @kspangsege can verify that the `ChangesetCooker` on `Sync::Client` is a different cooker than the one passed into `SyncHistory`.

Every `sync::Sync::History` contains a pointer to a cooker, or null when cooking is not enabled. Your cooker is called called from the internals of `sync::SyncHistory`.

Each session (`sync::Session`) has its own `sync::SyncHistory` object, and those history objects are created automatically as the sessions are created. A pointer to the cooker (or null) is passed via `sync::SyncHistory::Config` to the history constructor (`sync::make_sync_history()`).

If a session specific cooker pointer is specified via `sync::Session::Config`, then that pointer is passed to the history constructor. Otherwise if a client-wide cooker pointer is specified via `sync::Client::Config`, then that pointer is passed. Otherwise null is passed (cooking disabled).

In any case, neither the `sync::Client`, the `sync::Session`, or the `sync::SyncHistory` assumes ownership of any cooker. It is always the applications (object stores) responsibility to keep the cooker alive for sufficiently long.

Does this answer your question?